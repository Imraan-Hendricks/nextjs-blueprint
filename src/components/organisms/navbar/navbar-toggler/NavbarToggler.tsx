import { clsx } from 'clsx';
import { HiOutlineMenuAlt1, HiX } from 'react-icons/hi';
import { IconType } from 'react-icons/lib';

interface NavbarTogglerProps {
  isOpen: boolean;
  onClick: () => void;
}

export function NavbarToggler({ isOpen, onClick }: NavbarTogglerProps) {
  return (
    <div
      role='navbar-toggler'
      className={clsx(
        'group w-max cursor-pointer rounded border p-1.5',
        'border-gray-200 hover:bg-gray-200 active:bg-transparent',
        'dark:border-gray-700 dark:hover:bg-gray-700 dark:active:bg-transparent',
      )}
      onClick={onClick}>
      {isOpen && <IconWrapper Icon={HiX} testId='close-icon' />}
      {!isOpen && <IconWrapper Icon={HiOutlineMenuAlt1} testId='menu-icon' />}
    </div>
  );
}

interface IconWrapper {
  Icon: IconType;
  testId: 'close-icon' | 'menu-icon';
}

function IconWrapper({ Icon, testId }: IconWrapper) {
  return (
    <Icon
      data-testid={testId}
      className={clsx(
        'h-6 w-auto',
        'text-gray-500 group-hover:text-gray-800 group-active:text-gray-500',
        'dark:text-gray-400 group-hover:dark:text-gray-100 group-active:dark:text-gray-400',
      )}
    />
  );
}
