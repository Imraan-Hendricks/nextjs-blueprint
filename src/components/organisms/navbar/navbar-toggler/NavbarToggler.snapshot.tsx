import { cleanup, render } from '@testing-library/react';
import { NavbarToggler } from './NavbarToggler';

describe('NavbarToggler', () => {
  beforeAll(() => {
    cleanup();
  });

  afterEach(() => {
    cleanup();
  });

  test('renders with menu icon when isOpen is false', () => {
    const { container } = render(
      <NavbarToggler isOpen={false} onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with close icon when isOpen is true', () => {
    const { container } = render(
      <NavbarToggler isOpen={true} onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
});
