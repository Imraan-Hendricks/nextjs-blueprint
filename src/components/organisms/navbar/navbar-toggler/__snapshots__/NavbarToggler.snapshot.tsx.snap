// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarToggler renders with close icon when isOpen is true 1`] = `
<div>
  <div
    class="group w-max cursor-pointer rounded border p-1.5 border-gray-200 hover:bg-gray-200 active:bg-transparent dark:border-gray-700 dark:hover:bg-gray-700 dark:active:bg-transparent"
    role="navbar-toggler"
  >
    <svg
      aria-hidden="true"
      class="h-6 w-auto text-gray-500 group-hover:text-gray-800 group-active:text-gray-500 dark:text-gray-400 group-hover:dark:text-gray-100 group-active:dark:text-gray-400"
      data-testid="close-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`NavbarToggler renders with menu icon when isOpen is false 1`] = `
<div>
  <div
    class="group w-max cursor-pointer rounded border p-1.5 border-gray-200 hover:bg-gray-200 active:bg-transparent dark:border-gray-700 dark:hover:bg-gray-700 dark:active:bg-transparent"
    role="navbar-toggler"
  >
    <svg
      aria-hidden="true"
      class="h-6 w-auto text-gray-500 group-hover:text-gray-800 group-active:text-gray-500 dark:text-gray-400 group-hover:dark:text-gray-100 group-active:dark:text-gray-400"
      data-testid="menu-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16M4 12h8m-8 6h16"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
</div>
`;
